sources/packages/backend/apps/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -94,14 +100,24 @@ export class InstitutionLocationInstitutionsController extends BaseController {
       );
     }
 
-    // If the data is valid the location is saved to SIMS DB.
-    const createdInstitutionLocation = await this.locationService.saveLocation(
-      userToken.authorizations.institutionId,
-      dryRunSubmissionResult.data.data,
-      userToken.userId,
-    );
-
-    return { id: createdInstitutionLocation.id };
+    try {
+      // If the data is valid the location is saved to SIMS DB.
+      const createdInstitutionLocation =
+        await this.locationService.saveLocation(
+          userToken.authorizations.institutionId,
+          dryRunSubmissionResult.data.data,
+          userToken.userId,
+        );
+      return { id: createdInstitutionLocation.id };
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        if (error.name === DUPLICATE_INSTITUTION_LOCATION_CODE) {
+          throw new UnprocessableEntityException(
+            new ApiProcessError(error.message, error.name),
+          );
+        }
+      }
```

Comment:
We need to add a `throw ex` after the `if` block otherwise the method will fail silently.

Reply:
yes, good catch! Thanks!